examples/twittermap/web/public/javascripts/common/services.js

```diff
@@ -568,17 +592,56 @@ angular.module('cloudberry.common', ['cloudberry.mapresultcache'])
           case "sample":
             cloudberryService.commonTweetResult = result.value[0];
             break;
+          case "timeSeriesRequest":
+            var prefix = Object.freeze({
+                state: 'S',
+                county: 'C',
+                city: 'I'
+              });
+            const INVALID_VALUE = 0;
+                
+            var timeSeriesStore = new HashMap();
+            var geoIdSet = new Set(cloudberryService.parameters.geoIds);
+            for (var i = 0; i < result.value[0].length; i++) {
+              var currVal = result.value[0][i];
+              var geoIds = prefix[cloudberryService.parameters.geoLevel] + currVal[cloudberryService.parameters.geoLevel];
+              var values = timeSeriesStore.get(geoIds);
+              // First updates the store with geoIds that have results.
+              if (values !== undefined && values !== INVALID_VALUE) { // when one geoIds has more than one value
+                values.push({day:currVal["day"], count:currVal["count"]});
```

Comment:
As we discussed, the overhead of communication on entire String of `day` attribute for the bins might be large when number of records returned reach to thousands. If we can use offset # of days after the `startDate` as bin index, it might be more efficient.